src/App.tsx

```diff
@@ -1,14 +1,65 @@
-import React from 'react';
+import debounce from 'lodash.debounce';
+import { useMemo, useCallback, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { List } from './components/List/List';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setApplyQuerry] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+
+  const reset = () => {
+    setSelectedPerson('');
+  };
+
+  const handleValue = () => {
+    let value = query;
+
+    if (selectedPerson) {
+      value = selectedPerson;
+    }
+
+    return value;
+  };
+
+  const onSelected = (newPerson: string) => {
+    if (newPerson !== selectedPerson) {
+      setSelectedPerson(newPerson);
+    }
+  };
+
+  const findPerson = peopleFromServer.find(
+    person => person.name === selectedPerson,
+  );
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuerry = useCallback(
+    debounce(setApplyQuerry, 1000),
+    [],
+  );
+
+  const handlerQuery = (event:React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuerry(event.target.value);
+    reset();
+  };
+
+  const filteredPeople: Person[] = useMemo(() => {
+    return [...peopleFromServer].filter(
+      person => person.name.includes(appliedQuery.trim()),
+    );
+  },
+  [appliedQuery]);
```

Comment:
please debug this functions because the filtering is not working or working not properly:
![image](https://github.com/mate-academy/react_autocomplete/assets/102289654/51d2f445-1885-433c-8f26-222de32fc6e1)
![image](https://github.com/mate-academy/react_autocomplete/assets/102289654/2759f41d-e641-482b-99b3-1dbfa9526d7b)